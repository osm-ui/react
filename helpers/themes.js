import _ from 'lodash';
import { lighten } from 'polished';


export function buildDarkThemeConfig(baseConfig, colors, colorVariant) {
    const color1 = colors[`${colorVariant}1`];
    const color2 = colors[`${colorVariant}2`];
    const color3 = colors[`${colorVariant}3`];
    // const color4 = colors[`${colorVariant}4`];
    const color5 = colors[`${colorVariant}5`];

    return _.merge({}, baseConfig, {
        color: colors.white,
        backgroundColor: color2,
        borderColor: color1,
        loaderColor: colors.white,
        controlColor: color1,
        hoverControlColor: colors.white,

        column: {
            nav: {
                color: colors.white,
                backgroundColor: lighten(0.1, color2),
                hoverColor: colors.white,
                hoverBackgroundColor: color3,
            },
        },
        toolbar: {
            button: {
                color: colors.white,
                backgroundColor: color2,
                borderColor: color2,
                hoverBackgroundColor: color3,
                hoverBorderColor: color3,
                focusBackgroundColor: color3,
                focusBorderColor: color3,
                activeBackgroundColor: color3,
                activeBorderColor: color3,
            },
        },
        titlebar: {
            color: colors.white,
            backgroundColor: color2,
            borderColor: color2,

            button: {
                color: colors.white,
                backgroundColor: color2,
                borderColor: color2,
                hoverBackgroundColor: color3,
                hoverBorderColor: color3,
                focusBackgroundColor: color3,
                focusBorderColor: color3,
                activeBackgroundColor: color3,
                activeBorderColor: color3,
            },
        },
        form: {
            button: {
                color: colors.white,
                backgroundColor: color2,
                borderColor: color5,
                hoverBackgroundColor: color3,
                hoverBorderColor: color5,
                focusBackgroundColor: color3,
                focusBorderColor: color5,
                activeBackgroundColor: color3,
                activeBorderColor: color5,

                primary: {
                    color: color2,
                    backgroundColor: colors.white,
                    borderColor: colors.white,
                    hoverBackgroundColor: colors.lightGray5,
                    hoverBorderColor: colors.lightGray5,
                    focusBackgroundColor: colors.lightGray5,
                    focusBorderColor: colors.lightGray5,
                    activeBackgroundColor: colors.lightGray5,
                    activeBorderColor: colors.lightGray5,
                },
            },
        },
    });
}


export function buildLightThemeConfig(baseConfig, colors, colorVariant) {
    const color1 = colors[`${colorVariant}1`];
    const color2 = colors[`${colorVariant}2`];
    const color3 = colors[`${colorVariant}3`];
    const color4 = colors[`${colorVariant}4`];
    const color5 = colors[`${colorVariant}5`];

    return _.merge({}, baseConfig, {
        color: colors.anthracite1,
        backgroundColor: color2,
        borderColor: color1,
        loaderColor: colors.anthracite1,
        controlColor: color4,
        hoverControlColor: colors.anthracite1,

        column: {
            nav: {
                color: colors.anthracite1,
                backgroundColor: lighten(0.2, color2),
                hoverColor: colors.anthracite1,
                hoverBackgroundColor: color3,
            },
        },
        toolbar: {
            button: {
                color: colors.anthracite1,
                backgroundColor: color2,
                borderColor: color2,
                hoverBackgroundColor: color3,
                hoverBorderColor: color3,
                focusBackgroundColor: color3,
                focusBorderColor: color3,
                activeBackgroundColor: color3,
                activeBorderColor: color3,
            },
        },
        titlebar: {
            color: colors.anthracite1,
            backgroundColor: color2,
            borderColor: color2,

            button: {
                color: colors.anthracite1,
                backgroundColor: color2,
                borderColor: color2,
                hoverBackgroundColor: color3,
                hoverBorderColor: color3,
                focusBackgroundColor: color3,
                focusBorderColor: color3,
                activeBackgroundColor: color3,
                activeBorderColor: color3,
            },
        },
        form: {
            button: {
                color: colors.anthracite1,
                backgroundColor: color2,
                borderColor: color5,
                hoverBackgroundColor: color3,
                hoverBorderColor: color5,
                focusBackgroundColor: color3,
                focusBorderColor: color5,
                activeBackgroundColor: color3,
                activeBorderColor: color5,

                primary: {
                    color: color2,
                    backgroundColor: colors.anthracite1,
                    borderColor: colors.anthracite1,
                    hoverBackgroundColor: colors.lightGray5,
                    hoverBorderColor: colors.lightGray5,
                    focusBackgroundColor: colors.lightGray5,
                    focusBorderColor: colors.lightGray5,
                    activeBackgroundColor: colors.lightGray5,
                    activeBorderColor: colors.lightGray5,
                },
            },
        },
    });
}


export default {
    buildDarkThemeConfig,
    buildLightThemeConfig,
};
