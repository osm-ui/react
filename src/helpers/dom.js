export function isOverflowYHidden(element) {
  return window.getComputedStyle(element).overflowY !== 'hidden';
}
