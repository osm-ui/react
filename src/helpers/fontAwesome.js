export function makePrefix(style) {
  return `fa${style[0]}`;
}

export function makeName(name) {
  return `fa-${name}`;
}
