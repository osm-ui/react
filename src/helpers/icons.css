@import url('https://use.fontawesome.com/releases/v5.2.0/css/all.css');
