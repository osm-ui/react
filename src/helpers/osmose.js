export function formatDeletedTags(fix, osmTags) {
  return {
    ...fix,
    del: fix.del.map(deletedTag => {
      return osmTags.find(tag => tag.k === deletedTag);
    })
  };
}

export function formatOsmTags(data) {
  return data.reduce((acc, item) => {
    return {
      ...acc,
      [item.k]: item.v
    };
  }, {});
}

export function fixOsmTags(osmTags, fix) {
  const newOsm = {
    ...formatOsmTags(osmTags),
    ...formatOsmTags(fix.add),
    ...formatOsmTags(fix.mod)
  };

  fix.del.forEach(tag => Reflect.deleteProperty(newOsm, tag));

  return newOsm;
}
