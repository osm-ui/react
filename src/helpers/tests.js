import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';

import DefaultTheme from 'themes/Default';

export const snapshotWithElementChildren = (TestedElement, props) => {
  const result = shallow(
    <DefaultTheme>
      <TestedElement {...props}>
        <div />
      </TestedElement>
    </DefaultTheme>
  );

  expect(result.childAt(0)).toMatchSnapshot();
};

export const snapshotWithTextChildren = (TestedElement, props) => {
  const result = shallow(
    <DefaultTheme>
      <TestedElement {...props}>This is a text</TestedElement>
    </DefaultTheme>
  );

  expect(result.childAt(0)).toMatchSnapshot();
};
