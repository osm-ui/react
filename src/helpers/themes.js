import _ from 'lodash';
import { lighten, darken } from 'polished';
import { injectGlobal } from 'styled-components';

export function buildDarkThemeConfig(baseConfig, colors, colorVariant) {
  const color1 = colors[`${colorVariant}1`];
  const color2 = colors[`${colorVariant}2`];
  const color3 = colors[`${colorVariant}3`];
  // const color4 = colors[`${colorVariant}4`];
  const color5 = colors[`${colorVariant}5`];

  /* eslint-disable no-unused-expressions */
  injectGlobal`
    .osm-ui-react-marker-${colorVariant} {
      color: ${colors.white};

      #colorized, #colorized * {
        fill: ${color2} !important;
      }
    }
  `;
  /* eslint-enable */

  return _.merge({}, baseConfig, {
    color: colors.white,
    backgroundColor: color2,
    borderColor: color1,
    loaderColor: colors.white,
    controlColor: color1,
    hoverControlColor: colors.white,

    sidebar: {
      nav: {
        color: colors.white,
        backgroundColor: lighten(0.1, color2),
        hoverColor: colors.white,
        hoverBackgroundColor: color3
      }
    },
    toolbar: {
      button: {
        color: colors.white,
        backgroundColor: color2,
        borderColor: color2,
        hoverBackgroundColor: color3,
        hoverBorderColor: color3,
        focusBackgroundColor: color3,
        focusBorderColor: color3,
        activeBackgroundColor: color3,
        activeBorderColor: color3
      }
    },
    titlebar: {
      color: colors.white,
      backgroundColor: color2,
      borderColor: color2,

      button: {
        color: colors.white,
        backgroundColor: color2,
        borderColor: color2,
        hoverBackgroundColor: color3,
        hoverBorderColor: color3,
        focusBackgroundColor: color3,
        focusBorderColor: color3,
        activeBackgroundColor: color3,
        activeBorderColor: color3
      }
    },
    form: {
      button: {
        color: colors.white,
        backgroundColor: color2,
        borderColor: color5,
        hoverBackgroundColor: color3,
        hoverBorderColor: color5,
        focusBackgroundColor: color3,
        focusBorderColor: color5,
        activeBackgroundColor: color3,
        activeBorderColor: color5,

        primary: {
          color: color2,
          backgroundColor: colors.white,
          borderColor: colors.white,
          hoverBackgroundColor: colors.lightGray5,
          hoverBorderColor: colors.lightGray5,
          focusBackgroundColor: colors.lightGray5,
          focusBorderColor: colors.lightGray5,
          activeBackgroundColor: colors.lightGray5,
          activeBorderColor: colors.lightGray5
        }
      }
    }
  });
}

export function buildLightThemeConfig(baseConfig, colors, colorVariant) {
  const color1 = colors[`${colorVariant}1`];
  const color2 = colors[`${colorVariant}2`];
  const color3 = colors[`${colorVariant}3`];
  const color4 = colors[`${colorVariant}4`];
  const color5 = colors[`${colorVariant}5`];
  const darkColor1 = darken(0.27, color1);

  /* eslint-disable no-unused-expressions */
  injectGlobal`
        .osm-ui-react-marker-${colorVariant} {
            color: ${darkColor1};

            #colorized, #colorized * {
                fill: ${color2} !important;
            }
        }
    `;
  /* eslint-enable */

  return _.merge({}, baseConfig, {
    color: darkColor1,
    backgroundColor: color2,
    borderColor: color1,
    loaderColor: darkColor1,
    controlColor: color4,
    hoverControlColor: darkColor1,

    sidebar: {
      nav: {
        color: darkColor1,
        backgroundColor: color3,
        hoverColor: darkColor1,
        hoverBackgroundColor: lighten(0.14, color3)
      }
    },
    toolbar: {
      button: {
        color: darkColor1,
        backgroundColor: color2,
        borderColor: color2,
        hoverBackgroundColor: color3,
        hoverBorderColor: color3,
        focusBackgroundColor: color3,
        focusBorderColor: color3,
        activeBackgroundColor: color3,
        activeBorderColor: color3
      }
    },
    titlebar: {
      color: darkColor1,
      backgroundColor: color2,
      borderColor: color2,

      button: {
        color: darkColor1,
        backgroundColor: color2,
        borderColor: color2,
        hoverBackgroundColor: color3,
        hoverBorderColor: color3,
        focusBackgroundColor: color3,
        focusBorderColor: color3,
        activeBackgroundColor: color3,
        activeBorderColor: color3
      }
    },
    form: {
      button: {
        color: darkColor1,
        backgroundColor: color2,
        borderColor: color5,
        hoverBackgroundColor: color3,
        hoverBorderColor: color5,
        focusBackgroundColor: color3,
        focusBorderColor: color5,
        activeBackgroundColor: color3,
        activeBorderColor: color5,

        primary: {
          color: color2,
          backgroundColor: darkColor1,
          borderColor: darkColor1,
          hoverBackgroundColor: colors.lightGray5,
          hoverBorderColor: colors.lightGray5,
          focusBackgroundColor: colors.lightGray5,
          focusBorderColor: colors.lightGray5,
          activeBackgroundColor: colors.lightGray5,
          activeBorderColor: colors.lightGray5
        }
      }
    }
  });
}

export default {
  buildDarkThemeConfig,
  buildLightThemeConfig
};
