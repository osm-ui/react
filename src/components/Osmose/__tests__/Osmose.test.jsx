import 'jest-styled-components';
import { snapshotWithElementChildren } from 'helpers/tests';
import { Osmose } from 'index';

import fixData from '../../../../stories/mocks/osmose_fix.json';

describe('When using snapshots', () => {
  it('Should render with element children ', () =>
    snapshotWithElementChildren(Osmose, { data: fixData }));
});
