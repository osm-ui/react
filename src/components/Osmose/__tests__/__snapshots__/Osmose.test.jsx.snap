// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When using snapshots Should render with element children  1`] = `
.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen-Sans,Ubuntu,Cantarell,"Helvetica Neue",sans-serif;
}

.c2 {
  z-index: 1000;
  -webkit-transition: all 0.1s ease-out;
  transition: all 0.1s ease-out;
  left: 0;
  right: 0;
  font-weight: 500;
  color: #101113;
  border-color: #dbdcdd;
  border-style: solid;
  border-width: 0;
  background: #ffffff;
  box-shadow: 0 3px 4px -3px rgba(0,0,0,0.5),0 0 2px -1px rgba(0,0,0,0.4);
  text-align: center;
}

.c2.container-parent {
  position: absolute;
}

.c2.container-root {
  position: fixed;
}

.c2.position-top {
  top: 0;
  border-bottom-width: 1px;
}

.c2.position-bottom {
  bottom: 0;
  border-top-width: 1px;
}

.c2.xs {
  height: 3rem;
  font-size: 1.4rem;
  line-height: 3rem;
}

.c2.sm {
  height: 3.5rem;
  font-size: 1.6rem;
  line-height: 3.5rem;
}

.c2.md {
  height: 4rem;
  font-size: 1.8rem;
  line-height: 4rem;
}

.c2.lg {
  height: 4.5rem;
  font-size: 1.9rem;
  line-height: 4.5rem;
}

.c2.maximized {
  height: 100%;
  top: 0;
  bottom: 0;
  border-width: 0;
}

.c3 {
  margin-top: 3rem;
  cursor: pointer;
}

.c3:hover .infos {
  box-shadow: 0 3px 4px -3px rgba(0,0,0,0.5),0 0 2px -1px rgba(0,0,0,0.4);
}

.c3 .title {
  font-size: 1.4rem;
  padding-top: 0.2rem;
  text-align: center;
  color: #3b3f45;
  border-width: 2px;
  border-style: solid;
  border-color: #c9cacc;
  border-radius: 12px 12px 0 0;
  background-color: #ffffff;
}

.c3 .infos {
  padding: 0.7rem 1rem;
  padding-top: 1rem;
  border-width: 2px;
  border-style: solid;
  border-color: #c9cacc;
  border-radius: 0 0 12px 12px;
  border-top: none;
}

.c3 .infos .info {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0.2rem 0;
}

.c3 .infos .info.add {
  color: #348538;
}

.c3 .infos .info.add::before {
  content: '\\f067';
  font-family: FontAwesome;
  font-size: 1.3rem;
  line-height: 2.8rem;
  margin-right: 0.5rem;
}

.c3 .infos .info.mod {
  color: #de8c00;
}

.c3 .infos .info.mod::before {
  content: '\\f192';
  font-family: FontAwesome;
  font-size: 1.3rem;
  line-height: 2.8rem;
  margin-right: 0.5rem;
}

.c3 .infos .info.del {
  color: #c92d2d;
  text-decoration: line-through;
}

.c3 .infos .info.del::before {
  content: '\\f068';
  font-family: FontAwesome;
  font-size: 1.3rem;
  line-height: 2.7rem;
  margin-right: 0.5rem;
}

.c3 .infos .info .tag {
  -webkit-flex: 0 0 10rem;
  -ms-flex: 0 0 10rem;
  flex: 0 0 10rem;
  line-height: 2.7rem;
  font-size: 1.4rem;
  font-weight: bold;
}

.c3 .infos .info .value {
  font-size: 1.5rem;
  line-height: 2.7rem;
}

.c3.fix .title,
.c3.fix .infos,
.c3.new .title,
.c3.new .infos {
  border-color: #186fc5;
}

.c3.fix .title,
.c3.new .title {
  color: #ffffff;
  background-color: #2196f3;
}

.c1 {
  position: relative;
  color: black;
  height: 90vh;
  background-color: #ffffff;
}

.c1 > div {
  height: 100%;
  padding-bottom: 2rem;
  padding-top: 4.5rem;
  overflow-y: scroll;
}

.c1 h3 {
  text-align: center;
  margin-top: 1rem;
  font-size: 1.5rem;
}

@media (max-width:50rem) {
  .c3 .infos .info .tag {
    -webkit-flex: 0 0 30%;
    -ms-flex: 0 0 30%;
    flex: 0 0 30%;
  }
}

<span
  className="c0"
>
  <div
    className="c1"
  >
    <header
      className="sm position-top container-parent c2"
    >
      School, integration suggestion
    </header>
    <div>
      <h3>
        École primaire Louis Andrea 1, RUE EUTROPE MARIAN, 97122, BAIE MAHAULT (positioned: building, matching: manual)
      </h3>
      <div>
        <div
          className="osm c3"
          onClick={[Function]}
        >
          <div
            className="title"
          >
            Data in OSM
          </div>
          <div
            className="infos"
          >
            <div
              className="info"
            >
              <div
                className="tag"
              >
                amenity
              </div>
              <div
                className="value"
              >
                school
              </div>
            </div>
            <div
              className="info"
            >
              <div
                className="tag"
              >
                name
              </div>
              <div
                className="value"
              >
                Collège Maurice SATINEAU
              </div>
            </div>
          </div>
        </div>
        <div
          className="fix c3"
          onClick={[Function]}
        >
          <div
            className="title"
          >
            Suggestion n°1
          </div>
          <div
            className="infos"
          >
            <div
              className="info add"
            >
              <div
                className="tag"
              >
                ref:UAI
              </div>
              <div
                className="value"
              >
                9710194J
              </div>
            </div>
            <div
              className="info add"
            >
              <div
                className="tag"
              >
                school:FR
              </div>
              <div
                className="value"
              >
                primaire
              </div>
            </div>
            <div
              className="info add"
            >
              <div
                className="tag"
              >
                source
              </div>
              <div
                className="value"
              >
                data.gouv.fr:Éducation Nationale - 05/2016
              </div>
            </div>
          </div>
        </div>
        <div
          className="fix c3"
          onClick={[Function]}
        >
          <div
            className="title"
          >
            Suggestion n°2
          </div>
          <div
            className="infos"
          >
            <div
              className="info add"
            >
              <div
                className="tag"
              >
                ref:UAI
              </div>
              <div
                className="value"
              >
                9710194J
              </div>
            </div>
            <div
              className="info add"
            >
              <div
                className="tag"
              >
                school:FR
              </div>
              <div
                className="value"
              >
                primaire
              </div>
            </div>
            <div
              className="info add"
            >
              <div
                className="tag"
              >
                source
              </div>
              <div
                className="value"
              >
                data.gouv.fr:Éducation Nationale - 05/2016
              </div>
            </div>
            <div
              className="info mod"
            >
              <div
                className="tag"
              >
                name
              </div>
              <div
                className="value"
              >
                École primaire Louis Andrea 1
              </div>
            </div>
            <div
              className="info del"
            >
              <div
                className="tag"
              >
                amenity
              </div>
              <div
                className="value"
              >
                school
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</span>
`;
